src/main/java/graphql/execution/ExecutionStrategy.java

```diff
@@ -1024,4 +1016,25 @@ protected ExecutionStepInfo createExecutionStepInfo(ExecutionContext executionCo
                 .arguments(argumentValues)
                 .build();
     }
+
+    @NotNull
+    private static Supplier<ImmutableMapWithNullValues<String, Object>> getArgumentValues(ExecutionContext executionContext,
+                                                                                          List<GraphQLArgument> fieldArgDefs,
+                                                                                          List<Argument> fieldArgs) {
+        Supplier<ImmutableMapWithNullValues<String, Object>> argumentValues;
+        GraphQLCodeRegistry codeRegistry = executionContext.getGraphQLSchema().getCodeRegistry();
+        Supplier<ImmutableMapWithNullValues<String, Object>> argValuesSupplier = () -> {
+            Map<String, Object> resolvedValues = ValuesResolver.getArgumentValues(codeRegistry,
+                    fieldArgDefs,
+                    fieldArgs,
+                    executionContext.getCoercedVariables(),
+                    executionContext.getGraphQLContext(),
+                    executionContext.getLocale());
+
+            return ImmutableMapWithNullValues.copyOf(resolvedValues);
+        };
+        argumentValues = FpKit.intraThreadMemoize(argValuesSupplier);
+        return argumentValues;
+    }
+
```

Comment:
made a method - they are all the rage these days

Reply:
You're always up to date with the latest trends!